src/Agent.Listener/Diagnostics/DiagnosticsTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Diagnostics
+{
+    public class DiagnosticTests
+    {
+        public DiagnosticTests(ITerminal terminal)
+        {
+            m_terminal = terminal;
+            m_diagnosticTests = new List<IDiagnostic>()
+            {
+                new DnsTest(),
+                new PingTest(),
+            };
+        }
+
+        public void Execute()
+        {
+            bool result = true;
+            foreach(var test in m_diagnosticTests)
+            {
+                string testName = test.GetType().Name;
+                m_terminal.WriteLine(string.Format("*** Performing Test: {0} ***", testName));
+                if (!test.Execute(m_terminal))
```

Comment:
should we wrap text.Execute() in a try/catch just in case one of them blows up?

Reply:
Certainly doesn't hurt!  I think the individual tests of course should try to catch exceptions, but fi they don't we should catch them here.